Get rid of the mysql-python dependency

anselmrothe opened the pull request:
Until now, it was necessary to manually install the `mysql-python` package if one wanted to save the data in a [MySQL database](https://psiturk.readthedocs.io/en/latest/configure_databases.html?highlight=sql#using-a-self-hosted-mysql-database-recommended). Unfortunately, getting `mysql-python` properly installed was a nightmare. And `mysql-python` has hasn't been updated since 2014.

`pymysql` is a pure python alternative, which is [actively maintained](https://github.com/PyMySQL/PyMySQL) and which can be installed without problems. With `pymysql`, psiturk and your MySQL database can be happy again.

(Fixes #97)

gureckis commented:
nice @anselmrothe: do we want to may `pymysql` a requirement or optional install (as was `mysql-python`)?  it seems wise maybe to load as needed that was if mysql ever becomes a challenge again it doesn't stop new users (one of the advantages of the old approach)

gureckis commented:
also this change requires a few modifications to the docs that reference the older mysql library (e.g., https://psiturk.readthedocs.io/en/latest/install.html?highlight=mysql)

anselmrothe commented:
As far as I can tell, `pymysql` is a pretty safe bet (not more likely to become a reliability than the other packages in requirements.txt). The installation difficulties with `mysql-python` did not result from dealing with MySQL (which both packages do) but resulted from some C compilation difficulties which `pymysql` [got rid of](https://code.google.com/archive/p/pymysql/wikis/Goals.wiki).

So I'd have a slight preference for making `pymysql` a requirement but I let you make the judgement call here.

For the docs, I already made a search and there's only the part about [Debian systems](https://psiturk.readthedocs.io/en/latest/install.html?highlight=mysql#linux), where somebody who knows about Debian would need to check which packages need to be changed (I assume `python-mysqldb-dbg`).

Everything else in the docs can remain the same if we keep the pull request as it currently is. If we make `pymysql` optional, then we might want to update one or two places.

deargle commented:
It's just kind of bloat to require it. I don't use mysql when running on
heroku, but I do use postgres.

On Fri, Jun 29, 2018, 10:32 AM anselmrothe <notifications@github.com> wrote:

> As far as I can tell, pymysql is a pretty safe bet (not more likely to
> become a reliability than the other packages in requirements.txt). The
> installation difficulties with mysql-python did not result from dealing
> with MySQL (which both packages do) but resulted from some C compilation
> difficulties which pymysql got rid of
> <https://code.google.com/archive/p/pymysql/wikis/Goals.wiki>.
>
> So I'd have a slight preference for making pymysql a requirement but I
> let you make the judgement call here.
>
> For the docs, I already made a search and there's only the part about Debian
> systems
> <https://psiturk.readthedocs.io/en/latest/install.html?highlight=mysql#linux>,
> where somebody who knows about Debian would need to check which packages
> need to be changed (I assume python-mysqldb-dbg).
>
> Everything else in the docs can remain the same if we keep the pull
> request as it currently is. If we make pymysql optional, then we might
> want to update one or two places.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NYUCCL/psiTurk/pull/320#issuecomment-401407110>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABHsfTwbXx1CW23F3chWB5bR-ePFn6WDks5uBlangaJpZM4U8U5A>
> .
>

anselmrothe commented:
Make sense. I removed it from the requirements.

The new version checks whether `pymysql` is installed and prints a message otherwise (as for `mysql-python` before).

SQLAlchemy requires the database_url to begin with `mysql+pymysql://`. For backwards compatibility and convenience, the `+pymysql` gets added internally. So people don't have to update their existing config.txt files. The only other place beyond `db.py` that needed to be adjusted is `amt_services_wrapper.py`, which I updated.

The documentation is also updated.

gureckis commented:
looks like a worthwhile change, thanks @anselmrothe!
